index.d.ts

```diff
@@ -0,0 +1,75 @@
+// Type definitions for express-session 1.16
+// Project: https://github.com/expressjs/session
+// Definitions by: Hiroki Horiuchi <https://github.com/horiuchi>
+//                 Jacob Bogers <https://github.com/jacobbogers>
+//                 Naoto Yokoyama <https://github.com/builtinnya>
+//                 Ryan Cannon <https://github.com/ry7n>
+//                 HoldYourWaffle <https://github.com/HoldYourWaffle>
+
+import express from 'express'
+import node from 'events'
+
+
+
+declare global {
+  namespace Express {
+    interface Request { //inject additional properties on express.Request
+      session?: Session;
+      sessionID?: string;
+    }
+  }
+}
+
+
+
+interface SessionOptions {
+  secret: string | string[];
+
+  genid?(req: express.Request): string;
+  name?: string;
+  store?: Store;
+  cookie?: express.CookieOptions;
+  rolling?: boolean;
+  resave?: boolean;
+  proxy?: boolean;
+  saveUninitialized?: boolean;
+  unset?: string;
+}
+
+declare function session(options?: SessionOptions): express.RequestHandler;
+export default session;
+
+
+
+interface Session extends SessionData {
+  id: string;
+  cookie: SessionCookie;
+
+  regenerate(callback: (err: any) => void): void;
+  destroy(callback: (err: any) => void): void;
+  reload(callback: (err: any) => void): void;
+  save(callback: (err: any) => void): void;
```

Comment:
The callback is optional for this method, however you indicate that in typescript.